pkg/config/validator.go

```diff
@@ -222,42 +224,33 @@ func ConvertParams(find ValidatorFinderFunc, name string, params interface{}, st
 	return acfg, nil
 }
 
-// Decoder interface is used for injecting alternate decoders
-type Decoder interface {
-	Decode(raw interface{}) error
-}
-
-type newDecoderFn func(md *mapstructure.Metadata, dst interface{}) (Decoder, error)
-
-// NewDecoder creates a standard mapstructure decoder given metadata
-// but returns a Decoder interface
-func NewDecoder(md *mapstructure.Metadata, dst interface{}) (Decoder, error) {
-	return mapstructure.NewDecoder(
-		&mapstructure.DecoderConfig{Metadata: md, Result: dst},
-	)
+// convert map[interface{}] interface{} --> map[string] interface{}
+func convert(i interface{}) interface{} {
+	switch x := i.(type) {
+	case map[interface{}]interface{}:
+		m2 := map[string]interface{}{}
+		for k, v := range x {
+			m2[k.(string)] = convert(v)
+		}
+		return m2
+	case []interface{}:
+		for i, v := range x {
+			x[i] = convert(v)
+		}
+	}
+	return i
 }
 
 // Decode interprets src interface{} as the specified proto message.
-// optional newDecoderFn can be passed in, otherwise standard NewDecoder is used.
-func Decode(src interface{}, dst adapter.AspectConfig, strict bool, newDecoders ...newDecoderFn) (err error) {
-	var md mapstructure.Metadata
-	var d Decoder
-	newDecoder := NewDecoder
-	if len(newDecoders) > 0 {
-		newDecoder = newDecoders[0]
-	}
-
-	if d, err = newDecoder(&md, dst); err != nil {
-		return err
-	}
-
-	if err = d.Decode(src); err != nil {
+// if strict is true returns error on unknown fields.
+func Decode(src interface{}, dst adapter.AspectConfig, strict bool) (err error) {
+	var ba []byte
+	// src is potentially map[interface{}] interface{}
+	// unmarshaled by yaml. json expects map[string] interface{}
+	ba, err = json.Marshal(convert(src))
```

Comment:
or you can just pass strings from YAMLtoJSON to jsonpb since you serialize keys back to string anyways.